### PR TITLE
kata-containers: upgrade version 3.1.0 -> 3.2.0

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fixes: #6656

###### Change Log  

- kata-containers: fix bogus date
- kata-containers: upgrade version 3.1.0 -> 3.2.0
- kata-containers: cgmanifest: update entry
- kata-containers: drop patch setting dial_timeout=60 in favour of 45
- kata-containers: drop patches that are already upstreamed
- kata-containers: cosmetic fix to reorder patch numbers

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/45063880

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build w/ & w/o `RUN_CHECK=y`
- Pipeline build id: [PR-6657](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=444564&view=results)

### CODE REVIEW

**Review by anphel31 — APPROVED** [review]

File: SPECS/kata-containers/kata-containers.spec [hunk 1]
```diff
@@ -43,8 +43,8 @@
 
 Summary:        Kata Containers version 2.x repository
```
Comment (ms-mahuber) [inline SPECS/kata-containers/kata-containers.spec:44]: We should probably rename this, should get rid of 2.x

File: SPECS/kata-containers/kata-containers.spec [hunk 2]
```diff
@@ -53,14 +53,10 @@ Source1:        https://github.com/%{name}/%{name}/releases/download/%{version}/
...
-Patch5:         runtime-reduce-uvm-high-mem-footprint.patch
-Patch6:         drop-mut-for-variables-that-are-not-mutated.patch
-Patch7:         0001-osbuilder-Add-support-for-CBL-Mariner.patch
-Patch8:         0001-Append-systemd-kernel-cmdline-params-for-initrd.patch
+Patch1:         0004-Merged-PR-9806-Fix-enable_debug-for-hypervisor.clh.patch
+Patch2:         0005-Merged-PR-9956-shim-avoid-memory-hotplug-timeout.patch
+Patch3:         runtime-reduce-uvm-high-mem-footprint.patch
+Patch4:         0001-Append-systemd-kernel-cmdline-params-for-initrd.patch
 
 BuildRequires:  golang
```
Comment (ms-mahuber) [inline SPECS/kata-containers/kata-containers.spec:61]: Requires and Conficts sections:
- kata-agent, kata-ksm-throttler, kata-proxy, kata-runtime, kata-shim don't make sense to me. I don't think those packages exist. @danmihai1 do you remember?
- Requires: qemu-kvm-core: @danmihai1 should we leave this?

File: SPECS/kata-containers/kata-containers.spec (hunk 1 above)
Comment (ms-mahuber) [reply SPECS/kata-containers/kata-containers.spec:44]: Also, see another notion of 2.x below

File: SPECS/kata-containers/kata-containers.spec [hunk 3]
```diff
@@ -53,14 +53,10 @@ Source1:        https://github.com/%{name}/%{name}/releases/download/%{version}/
...
 Patch0:         0001-Merged-PR-9607-Allow-10-seconds-for-VM-creation-star.patch
-Patch1:         0002-Merged-PR-9671-Wait-for-a-possibly-slow-Guest.patch
-Patch2:         0003-Merged-PR-9805-Add-support-for-MSHV.patch
-Patch3:         0004-Merged-PR-9806-Fix-enable_debug-for-hypervisor.clh.patch
-Patch4:         0005-Merged-PR-9956-shim-avoid-memory-hotplug-timeout.patch
-Patch5:         runtime-reduce-uvm-high-mem-footprint.patch
-Patch6:         drop-mut-for-variables-that-are-not-mutated.patch
-Patch7:         0001-osbuilder-Add-support-for-CBL-Mariner.patch
-Patch8:         0001-Append-systemd-kernel-cmdline-params-for-initrd.patch
+Patch1:         0004-Merged-PR-9806-Fix-enable_debug-for-hypervisor.clh.patch
```
Comment (ms-mahuber) [inline SPECS/kata-containers/kata-containers.spec:56]: Patches: @danmihai1 , @dallasd1 @sprt - can you confirm we will still want/need to keep the remaining patches?

File: SPECS/kata-containers/runtime-reduce-uvm-high-mem-footprint.patch [hunk 4]
```diff
@@ -17,22 +17,26 @@ current mechanism of requesting resources through pod Limit annotations:
   ContainerCreating state
 
 Testing: Manual testing, creating pods with Limits and without limits, and with two containers where each container has a limit, tested with integration in a SPEC file where the config variables were set via environment variables via the make command
+
+Adapted by @mfrw from 3.1.0 to apply to 3.2.0
+
+Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>
```
Comment (ms-mahuber) [inline SPECS/kata-containers/runtime-reduce-uvm-high-mem-footprint.patch:23]: Please Microsoft address

File: SPECS/kata-containers/runtime-reduce-uvm-high-mem-footprint.patch [hunk 5]
```diff
@@ -17,22 +17,26 @@ current mechanism of requesting resources through pod Limit annotations:
   ContainerCreating state
 
 Testing: Manual testing, creating pods with Limits and without limits, and with two containers where each container has a limit, tested with integration in a SPEC file where the config variables were set via environment variables via the make command
+
```
Comment (ms-mahuber) [inline SPECS/kata-containers/runtime-reduce-uvm-high-mem-footprint.patch:20]: I have manually compared the current (CBL-Mariner/main) and new patch file (this branch/PR) against 3.1 and against 3.2. The changes look good. This patch has been properly ported.

File: SPECS/kata-containers/kata-containers.spec (hunk 2 above)
Comment (ms-mahuber) [reply SPECS/kata-containers/kata-containers.spec:61]: qemu-kvm-core: The dependency known to me is qemu-virtiofsd. Potentially we can refine the dependency.

File: SPECS/kata-containers/kata-containers.spec (hunk 3 above)
Comment (sprt) [reply SPECS/kata-containers/kata-containers.spec:56]: Confirm we don't need Patch2 and Patch7 either.

**Review by mfrw — COMMENTED** [review]
Superseded by:
- https://github.com/microsoft/CBL-Mariner/pull/6942

**Pull request closed without merging**